toucan_connectors/databricks/databricks_connector.py

```diff
@@ -0,0 +1,131 @@
+        connection_params = {
+            'Driver': '/opt/simba/spark/lib/64/libsparkodbc_sb64.so',  # This path must be correct when installing
+            # ODBC driver
+            'Host': '127.0.0.1' if self.host == 'localhost' else self.host,
+            'Port': self.port,
+            'HTTPPath': self.http_path,
+            'ThriftTransport': 2,
+            'SSL': 1,
+            'AuthMech': 3,
+            'UID': 'token',
+            'PWD': self.pwd.get_secret_value(),
+        }
+        return ';'.join(f'{k}={v}' for k, v in connection_params.items() if v is not None)
+
+    @staticmethod
+    def _get_details(index: int, status: Optional[bool]):
+        checks = ['Host resolved', 'Port opened', 'Connected to Databricks', 'Authenticated']
+        ok_checks = [(c, True) for i, c in enumerate(checks) if i < index]
+        new_check = (checks[index], status)
+        not_validated_checks = [(c, None) for i, c in enumerate(checks) if i > index]
+        return ok_checks + [new_check] + not_validated_checks
+
+    def get_status(self) -> ConnectorStatus:
+        # Check hostname
+        try:
+            self.check_hostname(self.host)
+        except Exception as e:
+            return ConnectorStatus(status=False, details=self._get_details(0, False), error=str(e))
+        # Check port
+        try:
+            self.check_port(self.host, self.port)
+        except Exception as e:
+            return ConnectorStatus(status=False, details=self._get_details(1, False), error=str(e))
+        try:
+            pyodbc.connect(self._build_connection_string())
+        except pyodbc.InterfaceError as e:
+            details = self._get_details(3, False)
+            details[2] = ('Connected to Databricks', True)
+            return ConnectorStatus(status=False, details=details, error=e.args[0])
+        except pyodbc.Error as e:
+            return ConnectorStatus(
+                status=False, details=self._get_details(2, False), error=e.args[0]
+            )
+        return ConnectorStatus(status=True, details=self._get_details(3, True), error=None)
+
+    def check_cluster_running(self) -> Dict[str, str]:
+        endpoint = f'https://{self.host}/api/2.0/clusters/get'
+        headers = {'login': 'token', 'password': self.pwd.get_secret_value()}
+        data = {'cluster_id': self.http_path.split('/')[-1]}
+        return requests.get(endpoint, headers=headers, json=data).json().get('state') == 'RUNNING'
+
+    def start_cluster(self) -> None:
+        endpoint = f'https://{self.host}/api/2.0/clusters/start'
+        headers = {'login': 'token', 'password': self.pwd.get_secret_value()}
+        data = {'cluster_id': self.http_path.split('/')[-1]}
+        resp = requests.post(endpoint, headers=headers, json=data)
+        if resp.status_code == 200:
+            logger.info('Databricks cluster started')
+        else:
+            logger.info(resp.json().get('message', 'Failed to start Databricks cluster'))
```

Comment:
It could be nice for the caller to have the information that the cluster did not start. Maybe raise an exception here ?

Reply:
yes, I add that.